scout/server/blueprints/variants/controllers.py

```diff
@@ -366,6 +378,13 @@ def parse_variant(
             variant_obj["compounds"], key=lambda compound: -compound["combined_score"]
         )
 
+        # check compound against current query rank score
+        for compound in variant_obj["compounds"]:
+            if (query_form and query_form.get("compound_rank_score") is not None) and (
```

Comment:
If you pass only the variant you could avoid the if (query_form and query_form.get("compound_rank_score") at least